modules/serverless-ossm-traffic-isolation-securing-the-service-mesh.adoc

```diff
@@ -0,0 +1,257 @@
+  rules:
+    - from:
+        - source:
+            namespaces: [ "knative-eventing" ]
+            principals: [ "cluster.local/ns/knative-eventing/sa/kafka-controller" ]
+      to:
+        - operation:
+            methods: [ "GET" ]
+---
+apiVersion: security.istio.io/v1beta1
+kind: AuthorizationPolicy
+metadata:
+  name: allow-probe-kafka-channel-receiver
+  namespace: knative-eventing
+spec:
+  action: ALLOW
+  selector:
+    matchLabels:
+      app.kubernetes.io/component: "kafka-channel-receiver"
+  rules:
+    - from:
+        - source:
+            namespaces: [ "knative-eventing" ]
+            principals: [ "cluster.local/ns/knative-eventing/sa/kafka-controller" ]
+      to:
+        - operation:
+            methods: [ "GET" ]
+---
+apiVersion: security.istio.io/v1beta1
+kind: AuthorizationPolicy
+metadata:
+  name: allow-traffic-to-activator
+  namespace: knative-serving
+spec:
+  selector:
+    matchLabels:
+      app: activator
+  action: ALLOW
+  rules:
+    - from:
+        - source:
+            namespaces: [ "knative-serving", "istio-system" ]
+---
+apiVersion: security.istio.io/v1beta1
+kind: AuthorizationPolicy
+metadata:
+  name: allow-traffic-to-autoscaler
+  namespace: knative-serving
+spec:
+  selector:
+    matchLabels:
+      app: autoscaler
+  action: ALLOW
+  rules:
+    - from:
+        - source:
+            namespaces: [ "knative-serving" ]
+----
++
+These policies restrict the access rules for the network communication between {ServerlessProductShortName} system components. Specifically, they enforce the following rules:
++
+--
+* Deny all traffic that is not explicitly allowed in `knative-serving` and `knative-eventing`
+* Allow traffic from `istio-system` and `knative-serving` to activator
+* Allow traffic from `knative-serving` to autoscaler
+* Allow health probes for Apache Kafka components in `knative-eventing`
+* Allow internal traffic for channel based brokers in `knative-eventing`
+--
+
+. Apply the authorization policy configuration:
++
+[source,terminal]
+----
+$ oc apply -f knative-default-authz-policies.yaml
+----
+
+. Define which OpenShift projects can communicate with each other. For that, every OpenShift project of a tenant needs:
++
+--
+* One `AuthorizationPolicy` limiting directly incoming traffic to the tenants OpenShift project
+* One `AuthorizationPolicy` limiting incoming traffic via the activator component of OpenShift Serverless that runs in the `knative-serving` OpenShift project
+* One `AuthorizationPolicy` allowing Kubernetes to call `PreStopHooks` on Knative Services
+--
++
+Instead of creating these policies by hand, run the `helm` utility to create the necessary resources for each tenant:
++
+.Example configuration for `team alpha`
+[source,terminal]
+----
+$ helm template oci://quay.io/openshift-knative/knative-istio-authz-onboarding --version 1.31.0 --set "name=team-alpha" --set "namespaces={team-alpha-1,team-alpha-2}" > team-alpha.yaml
```

Comment:
There will be a productized version of the helm chart. So this probably needs to change to that URL. cc @Kaustubh-pande

Reply:
Also ideally, the version (1.31.0) is from a parameter, so that it automatically matches the version we ship. Otherwise you'll have to update this on every release.

Reply:
Now that the helm chart was published, I think the right way to install it is:
```
$ helm repo add openshift-helm-charts https://charts.openshift.io/
$ helm template openshift-helm-charts/redhat-knative-istio-authz --version 1.31.0 --set "name=team-alpha" --set "namespaces={team-alpha-1,team-alpha-2}" > team-alpha.yaml
```
The first command will be done only once. The second command will be also called later for "team bravo"

The `oci://` stuff was used for developemtn purporses.

Reply:
We use/test it here: https://github.com/openshift-knative/serverless-operator/blob/main/hack/generate/mesh-auth-policies.sh#L29